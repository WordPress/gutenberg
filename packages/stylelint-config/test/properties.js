/**
 * Internal dependencies
 */
const utils = require( './utils' );
const getStylelintResult = utils.getStylelintResult;

describe( 'flags no warnings with valid properties css', () => {
	let result;

	beforeEach( () => {
		result = getStylelintResult( './properties-valid.css' );
	} );

	it( 'did not error', () => {
		return result.then( ( data ) => expect( data.errored ).toBeFalsy() );
	} );

	it( 'flags no warnings', () => {
		return result.then( ( data ) =>
			expect( data.results[ 0 ].warnings ).toHaveLength( 0 )
		);
	} );
} );

describe( 'flags warnings with invalid properties css', () => {
	let result;

	beforeEach( () => {
		result = getStylelintResult( './properties-invalid.css' );
	} );

	it( 'did error', () => {
		return result.then( ( data ) => expect( data.errored ).toBeTruthy() );
	} );

	it( 'flags correct number of warnings', () => {
		return result.then( ( data ) =>
			expect( data.results[ 0 ].warnings ).toHaveLength( 7 )
		);
	} );

	it( 'snapshot matches warnings', () => {
		return result.then( ( data ) =>
			expect( data.results[ 0 ].warnings ).toMatchSnapshot()
		);
	} );
} );
