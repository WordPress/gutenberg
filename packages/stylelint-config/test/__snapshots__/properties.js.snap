// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flags warnings with invalid properties css snapshot matches warnings 1`] = `
[
  {
    "column": 13,
    "endColumn": 14,
    "endLine": 2,
    "line": 2,
    "rule": "@stylistic/color-hex-case",
    "severity": "error",
    "text": "Expected "#FFFFFF" to be "#ffffff" (@stylistic/color-hex-case)",
  },
  {
    "column": 13,
    "endColumn": 14,
    "endLine": 2,
    "line": 2,
    "rule": "@stylistic/declaration-colon-space-after",
    "severity": "error",
    "text": "Expected single space after ":" with a single-line declaration (@stylistic/declaration-colon-space-after)",
  },
  {
    "column": 17,
    "endColumn": 19,
    "endLine": 4,
    "line": 4,
    "rule": "@stylistic/unit-case",
    "severity": "error",
    "text": "Expected "PX" to be "px" (@stylistic/unit-case)",
  },
  {
    "column": 13,
    "endColumn": 20,
    "endLine": 2,
    "line": 2,
    "rule": "color-hex-length",
    "severity": "error",
    "text": "Expected "#FFFFFF" to be "#FFF" (color-hex-length)",
  },
  {
    "column": 2,
    "endColumn": 8,
    "endLine": 5,
    "line": 5,
    "rule": "declaration-block-no-shorthand-property-overrides",
    "severity": "error",
    "text": "Unexpected shorthand "margin" after "margin-left" (declaration-block-no-shorthand-property-overrides)",
  },
  {
    "column": 2,
    "endColumn": 7,
    "endLine": 6,
    "line": 6,
    "rule": "property-no-unknown",
    "severity": "error",
    "text": "Unexpected unknown property "argin" (property-no-unknown)",
  },
  {
    "column": 11,
    "endColumn": 16,
    "endLine": 3,
    "line": 3,
    "rule": "value-keyword-case",
    "severity": "error",
    "text": "Expected "BLOCK" to be "block" (value-keyword-case)",
  },
]
`;
